src/datastructures/nalgebra.rs

```diff
@@ -20,16 +36,40 @@ impl From<Graph> for NAMatrix {
                 res[(i, j)] = x;
             }
         }
-        res
+        NAMatrix(res)
     }
 }
 
 impl AdjacencyMatrix for NAMatrix {
+    fn from_dim(dim: usize) -> Self {
+        NAMatrix(DMatrix::from_element(dim, dim, f64::INFINITY))
```

Comment:
are you sure that this is true?
It seems that if I create with `from_dim` and then set each vertices through `set`, the diagonal would be `f64::INFINITY`, which is wrong as it should be 0.

Reply:
You are right, I think resolved it now